pandas/core/arrays/categorical.py

```diff
@@ -523,6 +527,11 @@ def _from_inferred_categories(cls, inferred_categories, inferred_codes,
                 cats = to_datetime(inferred_categories, errors='coerce')
             elif is_timedelta64_dtype(dtype.categories):
                 cats = to_timedelta(inferred_categories, errors='coerce')
+            elif dtype.categories.is_boolean():
```

Comment:
this does NOT belong here at all, you are mixing up parsing with setting the categories. you can do this via a separate function *inside* parser.pyx

Reply:
I think there were issues with that (either nogil, or having to share code with the Python parser).

Reply:
well it certainly doesn’t belong here
this is conflating parsing with categories

Reply:
I see now that this is called outside a nogil block, so a standalone method for that should be fine. If you want to open an issue though, because I don't think it should hold up the release.

Reply:
this isn't necessary for the RC. this needs to be fixed in this PR.